crypto/s2n_certificate.h

```diff
@@ -53,7 +53,7 @@ typedef enum {
     S2N_CERT_TYPE_ECDSA_FIXED_ECDH = 66,
 } s2n_cert_type;
 
-struct s2n_cert_public_key {
```

Comment:
It's confusing to have both `struct s2n_cert` and `struct s2n_cert_chain_and_key`. The first seems to only be used for client auth and the latter for the server cert. Any way we can unify these? Or remove one of them? `cert_type` can be a bit misleading. It actually describes the key type.